site/functions/server/main.go

```diff
@@ -0,0 +1,287 @@
+		}
+		out.Data.Plugins = append(out.Data.Plugins, pi)
+	}
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func fetchPlugins(entries []*github.RepositoryContent) ([]*krew.Plugin, error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var (
+		mu     sync.Mutex
+		out    []*krew.Plugin
+		retErr error
+	)
+
+	queue := make(chan string)
+	var wg sync.WaitGroup
+
+	for i := 0; i < urlFetchBatchSize; i++ {
+		wg.Add(1)
+		go func(j int) {
+			defer wg.Done()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case url, ok := <-queue:
+					if !ok {
+						return
+					}
+					p, err := readPlugin(url)
+					if err != nil {
+						retErr = err
+						cancel()
+						return
+					}
+					mu.Lock()
+					out = append(out, p)
+					mu.Unlock()
+				}
+			}
+		}(i)
+	}
+
+	for _, v := range entries {
+		url := v.GetDownloadURL()
+		select {
+		case <-ctx.Done():
+			break
+		case queue <- url:
+		}
+	}
+
+	close(queue)
+	wg.Wait()
+
+	return out, retErr
+}
+
+func readPlugin(url string) (*krew.Plugin, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err,"failed to get %s", url)
+	}
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+	var v krew.Plugin
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err,"failed to read file %s: %w", url)
+	}
+
+	if err = yaml.Unmarshal(b, &v); err != nil {
+		return nil, errors.Wrapf(err,"failed to parse plugin manifest for %s", url)
+	}
+	return &v, nil
+}
+
+func findRepo(homePage string) string {
+	if matches := githubRepoPattern.FindStringSubmatch(homePage); matches != nil {
+		return matches[1]
+	}
+
+	knownHomePages := map[string]string{
+		`https://krew.sigs.k8s.io/`:                                  "kubernetes-sigs/krew",
+		`https://sigs.k8s.io/krew`:                                   "kubernetes-sigs/krew",
+		`https://kubernetes.github.io/ingress-nginx/kubectl-plugin/`: "kubernetes/ingress-nginx",
+		`https://kudo.dev/`:                                          "kudobuilder/kudo",
+		`https://kubevirt.io`:                                        "kubevirt/kubectl-virt-plugin",
+		`https://popeyecli.io`:                                       "derailed/popeye",
+		`https://soluble-ai.github.io/kubetap/`:                      "soluble-ai/kubetap",
+	}
+	return knownHomePages[homePage]
+}
+
+func main() {
+	port := flag.Int("port", -1, "specify a port to use http rather than AWS Lambda")
+	flag.Parse()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/.netlify/functions/api/pluginCount", pluginCountHandler)
+	mux.HandleFunc("/.netlify/functions/api/plugins", pluginsHandler)
+	// To debug locally, you can run this server with -port=:8080 and run "hugo serve" and uncomment this:
+	mux.Handle("/", httputil.NewSingleHostReverseProxy(&url.URL{Scheme: "http", Host: "localhost:1313"}))
```

Comment:
The comment reads like this should be commented out?

Anyway, I think it'd be much nicer like that, then no commenting is necessary:
```go
if *port > -1 {
   ...
}
```

Reply:
It can be commented out but I left it on just in case. You're right.